nengo_loihi/hardware/builder.py

```diff
@@ -379,11 +379,13 @@ def build_input(nxsdk_core, core, spike_input, compartment_idxs):
     nxsdk_board.spike_inputs[spike_input] = loihi_input
 
     # add any pre-existing spikes to spikegen
-    for t in spike_input.spike_times():
+    times = spike_input.spike_times()
+    if len(times) > 0:
```

Comment:
Was this assert a big time sink? The only downside to this is that `times` can't be a generator now, but if this is working, then I guess it wasn't a generator anyway, so we can always change accordingly if we want it to be a generator in the future.

Reply:
Since you made changes here for speed reasons I assumed it was a critical loop, so any savings would be worth it. It would probably be good for it to be a generator, and there are other ways to figure out if there are any spikes without doing `len`, so I'll look at this in more detail.

Reply:
Resolved in dd4afcea.